uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/bookingrequest/BookingRequestPresenter.kt

```diff
@@ -129,12 +126,19 @@ class BookingRequestPresenter(view: BookingRequestMVP.View,
     }
 
     override fun setBookingFields(allFieldsValid: Boolean) {
-        if (KarhooUISDKConfigurationProvider.isGuest()) {
-            view?.showGuestBookingFields()
-            setBookingEnablement(allFieldsValid)
-        } else {
-            view?.showAuthenticatedUserBookingFields()
-            setBookingEnablement(true)
+        when {
```

Comment:
Might be worth getting the auth method and using the when against that so that we're not making duplicate calls to retrieve it?